chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -118,6 +120,7 @@ extends HasId {
   private[chisel3] val _commands = ArrayBuffer[Command]()
   private[core] val _ids = ArrayBuffer[HasId]()
   Builder.currentModule = Some(this)
+  Builder.whenDepth = 0
```

Comment:
I added the check. I am of the opinion that whenDepth should be set in the Module itself to keep all side effects of creating a Module in the same place since currentModule, etc. are all set there. I know they *have* to be set inside of the Module but I think keeping them in the same place is more consistent.